[GaaneshT] iP

GaaneshT opened the pull request:
### Malt

> “Your mind is for having ideas, not holding them.” – David Allen [(source)](source_link_here)

DukePro frees your mind of having to remember things you need to do. It's,

- text-based  
- easy to learn  
- **FAST _SUPER_ FAST** to use  

All you need to do is,

1. download it from [here](download_link_here).  
2. double-click it.  
3. add your tasks.  
4. let it manage your tasks for you 😉  

And it is **FREE!**  

### Features:
- [x] Managing tasks  
- [x] Managing deadlines (coming soon)  
- [ ] Reminders (coming soon)  

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

GaaneshT committed:
Rename to Malt and completed Level-0

GaaneshT committed:
Created more functions and completed Level-1

GaaneshT committed:
Completed Level-2

GaaneshT committed:
Completed Level-3, added Task class.

GaaneshT committed:
Added Todo,Event and Deadline functionality. Completed Level-4

GaaneshT committed:
Added test cases, completed A-TextUiTesting

GaaneshT committed:
Added MaltExceptions, Completed Level-5

GaaneshT committed:
Added delete functionality, Completed Level-6

GaaneshT committed:
Implement file saving for Level-7

GaaneshT committed:
Merge Level-7 feature

GaaneshT committed:
Implement Level-8: store deadlines as LocalDate

GaaneshT committed:
Merge branch-Level-8 into master

GaaneshT committed:
Refactor code to more OOP-style

GaaneshT committed:
Refactor code into packages

GaaneshT committed:
Merge branch 'add-gradle-support'

GaaneshT committed:
change gradle.build to include malt

Jeremiah33-3 commented on src/main/java/malt/parser/Parser.java:
```diff
@@ -0,0 +1,224 @@
+        switch (command) {
+            case "bye":
+                // User wants to exit
+                return handleBye(ui);
```
just a suggestion! perhaps the naming of the methods can be made clearer if the word `Command` is specified.
```java
...
return handleByeCommand(ui);
...
return handleListCommand(ui);
...
```

GaaneshT replied:
Thanks for suggestion, I'll be sure to amend it!

Jeremiah33-3 commented on src/main/java/malt/parser/Parser.java:
```diff
@@ -0,0 +1,224 @@
+    // --------------------------------------------------
+
+    private static boolean handleBye(Ui ui) {
+        ui.showGoodbye();
```
Similarly, might be easier for readers to understand if this method name is changed to:
```java
showGoodbyeMessage();
```

GaaneshT replied:
Good point, thanks!

Jeremiah33-3 commented on src/main/java/malt/parser/Parser.java:
```diff
@@ -0,0 +1,224 @@
+    private static void handleDelete(String arg, TaskList tasks, Ui ui, Storage storage) throws MaltException {
+        try {
+            int index = Integer.parseInt(arg);
+            Task removed = tasks.removeTask(index - 1);
```
Here, `removed` might be mildly ambiguous, `removedTask` might be easier to understand!

GaaneshT replied:
I agree, will change it. Thanks!

Jeremiah33-3 commented on src/main/java/malt/parser/Parser.java:
```diff
@@ -0,0 +1,224 @@
+
+    private static void handleEvent(String arg, TaskList tasks, Ui ui, Storage storage) throws MaltException {
+        // e.g.: "project meeting /from Monday 2pm /to 4pm"
+        String[] fromSplit = arg.split("/from", 2);
```
emmm `fromSplit` sounds a bit confusing as to what this array is supposed to represent! might there be a clearer naming?

GaaneshT replied:
I thought fromSplit would be good since it like splits it at the "from" field so hence fromSplit. But you make a good point and I can see how it would be confusing. Will fix it

Jeremiah33-3 commented on src/main/java/malt/ui/Ui.java:
```diff
@@ -0,0 +1,105 @@
+                        /::\\____\\                /::\\    \\                /::\\____\\    /::\\    \\       \s
+                       /::::|   |               /::::\\    \\              /:::/    /    \\:::\\    \\      \s
+                      /:::::|   |              /::::::\\    \\            /:::/    /      \\:::\\    \\     \s
+                     /::::::|   |             /:::/\\:::\\    \\          /:::/    /        \\:::\\    \\    \s
```
this is actually so cool 😮

GaaneshT replied:
:D

Jeremiah33-3 reviewed (commented):
Overall, good work!

GaaneshT committed:
Add JUnit tests for Parser and deadline

GaaneshT committed:
Complete Level 9: Implement find functionality

GaaneshT committed:
Merge branch-Level-9: Implement find functionality

GaaneshT committed:
Add JavaDoc comments to non-private classes and methods

GaaneshT committed:
Merge branch-A-JavaDoc: Add JavaDoc comments

GaaneshT committed:
Apply coding standard changes

GaaneshT committed:
Merge branch-A-CodingStandard: Apply coding standard changes

GaaneshT committed:
Javafx complete. On branch

GaaneshT committed:
Merging branch-level-10 into master

GaaneshT committed:
Add FXML functionality.

GaaneshT committed:
fix: Add assertions to enforce key assumptions

The code currently lacks explicit checks for key assumptions, such as valid task indices, non-null user input, and properly initialized objects. This increases the risk of silent failures and unexpected behavior.

Without assertions, invalid states may go undetected during development, making debugging harder. Ensuring these assumptions hold improves code reliability and maintainability.

- Add assertions to `TaskList.java` to verify task indices before access or removal.
- Add assertions in `Parser.java` to confirm command input, tasks, UI, and storage are valid.
- Ensure `Storage.java` checks that file paths are properly set.
- Enforce that `MaltChatbot.java` does not return null or empty responses.
- Validate `Task.java`, `Deadline.java`, and `Event.java` to guarantee non-empty descriptions and valid time inputs.

GaaneshT committed:
Add assert cases

GaaneshT committed:
Merge branch-A-Assertions: Add assertions to enforce assumptions

GaaneshT committed:
Improve code quality based on review guidelines

GaaneshT committed:
Merge pull request #2 from GaaneshT/branch-A-CodeQuality

Improve code quality based on review guidelines

GaaneshT committed:
Add FriendlierSyntax function, clear function

GaaneshT committed:
Reformat code to apply styling rules

GaaneshT committed:
Improve flexible event parsing for

GaaneshT committed:
Fix build.gradlew runtime path for javaFX

GaaneshT committed:
Add UserGuide and fix bye command

GaaneshT committed:
Fix storage issue when data folder is not found

GaaneshT committed:
Add acknowledgements in README

GaaneshT committed:
Update javaFX GUI to disable resize
